turbopack/crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -385,6 +392,100 @@ impl SourceMap {
         Ok(token.cell())
     }
 
+    /// Traces a generated line/column into an mapping token representing either
+    /// synthetic code or user-authored original code.
+    #[turbo_tasks::function]
+    pub async fn lookup_token_and_source(
```

Comment:
You copied all the code from above method. Avoid duplication and move that into a shared method.